tools/Evm/Evm.Test/T8nTests.cs

```diff
@@ -0,0 +1,260 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using Evm.T8n;
+using Evm.T8n.JsonTypes;
+using Nethermind.Logging;
+using Nethermind.Serialization.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Evm.Test;
+
+public class InputParams(string basedir, string alloc, string env, string txs, string stateFork, string? stateReward = null)
+{
+    public readonly string Alloc = Path.Combine(basedir, alloc);
+    public readonly string Env = Path.Combine(basedir, env);
+    public readonly string Txs = Path.Combine(basedir, txs);
+    public readonly string StateFork = stateFork;
+    public readonly string? StateReward = stateReward;
+}
+
+public class OutputParams(string? alloc = null, string? result = null, string? body = null)
+{
+    public readonly string? Alloc = alloc;
+    public readonly string? Result = result;
+    public readonly string? Body = body;
+}
+
+public class T8nTests
+{
+    private readonly EthereumJsonSerializer _ethereumJsonSerializer = new();
+
+    [Test]
+    public void Test1()
+    {
+        Execute(
+            new InputParams("testdata/1/", "alloc.json", "env.json", "txs.json", "Frontier+1346"),
+            new OutputParams(alloc: "stdout", result: "stdout"),
+            expectedExitCode: 3);
+    }
+
+    [Test]
+    public void Test2()
+    {
+        Execute(
+            new InputParams("testdata/1/", "alloc.json", "env.json", "txs.json", "Byzantium"),
+            new OutputParams(alloc: "stdout", result: "stdout"),
+            expectedExitCode: 0,
+            expectedOutputFile: "testdata/1/exp.json");
+    }
+
+    [Test]
+    public void Test3()
+    {
+        Execute(
+            new InputParams("testdata/3/", "alloc.json", "env.json", "txs.json", "Berlin"),
+            new OutputParams(alloc: "stdout", result: "stdout"),
+            expectedExitCode: 0,
+            expectedOutputFile: "testdata/3/exp.json");
+    }
+
+    [Test]
+    public void Test4()
+    {
+        Execute(
+            new InputParams("testdata/4/", "alloc.json", "env.json", "txs.json", "Berlin"),
+            new OutputParams(alloc: "stdout", result: "stdout"),
+            expectedExitCode: 4);
+    }
+
+    [Test]
+    public void Test5()
+    {
+        Execute(
+            new("testdata/5/", "alloc.json", "env.json", "txs.json", "Byzantium", "0x80"),
+            new OutputParams(alloc: "stdout", result: "stdout"),
+            expectedExitCode: 0,
+            expectedOutputFile: "testdata/5/exp.json");
+    }
+
+    [Test]
+    public void Test6()
```

Comment:
Can you name tests in some sensible way? I see it is based on `testdata` so `Test13_txs_London`?